### PR TITLE
Trip weather serializer

### LINKED ISSUE #6: README Framework
- [x] Table of Contents
- [x] What it Does
- [x] How To Install
- [x] Dependencies
- [x] Testing
- [x] Schema
- [x] Learning Goals
- [x] Licences/Badges
- [x] Resources
- [x] Contact Collaborators
- [x] Acknowledgements

### LINKED ISSUE #11: BE Trip Poro
As a BE user, when I visit SearchController, a TripFacade is called.
Inside the TripFacade, Trip poros are made

The Trip Poro
would include:
- [x] id (for the trip)
- [x] origin city 
- [x] destination_city
- [x] origin_iata
- [x] destinationa_iata
- [x] latitude
- [x] longitude
- [x] price
- [x] departure_datetime
- [x] arrival_datetime
- [x] r_departure_datetime
- [x] r_arrival_datetime
- [x] booking_link
- [x] flight_id
- [x] min_f
- [x] max_f
- [x] min_c
- [x] max_c
- [x] day_feels_like_c
- [x] day_feels_like_f
- [x] description
- [x] sky_coverage
- [x] date (weather date)

- [ ] Happy/sad path testing

- [ ] Update to README (FE screenshots, BE API endpoint documented)

### LINKED ISSUE #14: BE Trip Facade
Trip facade that returns an array of `Trip` objects to eventually pass to `TripSerializer`.

- [ ] Happy/sad path testing

### LINKED ISSUE #16: BE Weather & Flight Services
Weather and flight service classes to connect to micro-services and fetch data.

- [x] Happy/sad path testing

- [x] Update to README (FE screenshots, BE API endpoint documented)

### PR DESCRIPTION
**Pull request by GeorgieGirl24**
closes #11 
closes #6 
closes #14 

- [ ] Updated Files:
    * app/poros
    * app/serializers
    * app/services
    * app/controllers
    * app/facades
    * spec/api/v1/search/index
    * spec/facades
    * spec/poros
- [ ] Repo Representation: Front End, Back End, Amedaus, Open Weather, Google 

     * Back End
     
- [x] Sad Path Tested

- [x] Happy Path Tested

- [x] All Tests are passing
   * Yes, but....
   * In `spec/api/v1/search/index_spec ln: 32` test is failing. We ran this in the Chrome terminal and it ACTUALLY works.
     The expectation is that if you leave a date empty you will get an error:
```
      {
        "error": "Invalid Data",
        "status": 400
      }
```
    However, we were unable to make it pass in RSpec

- [x] Updated README.md with added information

- [ ] Collabotors:
   * Jesse Mellinger
   * George Soderholm

### CODE REVIEW

**Comment by cpfergus1** [comment]
Travis is failing on Faraday connection, is webmock implemented for this branch?

**Comment by cpfergus1** [comment]
Fixed Travis ENV variables - looks good now!
Make a card on your failing test that needs a second look at.

**Review by cpfergus1 — APPROVED** [review]

File: app/facades/trip_facade.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    if flights[:message].nil?
+      flights[:data].each_with_index do |flight, index|
+        lat = flight[:attributes][:latitude]
+        lon = flight[:attributes][:longitude]
+        weather = WeatherService.get_weather(lat, lon)
+        trips << Trip.new(flight, weather, index)
+      end
+      trips #return value: an array of Trip Objects
+    else
+      flights
```
Comment (stillsheryl) [inline app/facades/trip_facade.rb:14]: Great job! I'm a bit confused here. What's the reason for returning `flights` here?

**Review by stillsheryl — APPROVED** [review]

File: app/facades/trip_facade.rb (hunk 1 above)
Comment (JesseMellinger) [reply app/facades/trip_facade.rb:14]: The variable name `flights` is a bit misleading here. `flights` will either have a `:data` key with an array of flight hashes or a `:message` key containing an error. So if `flights[:message]` is nil we create `Trip` objects else we return the hash containing the error.

A better variable here would probably be `response`.

Comment (stillsheryl) [reply app/facades/trip_facade.rb:14]: OK, thank you for the clarification!

**Review by Todd-Estes — APPROVED** [review]

**Review by SageOfCode — APPROVED** [review]
Looks ready to merge!

**Pull request merged**